Add method for topological sorting

fsobanski opened the pull request.

Commit:
Add method for topological sorting

fsobanski commented:
Would you consider accepting this PR? It's basically just an implementation of [this](https://en.wikipedia.org/wiki/Topological_sorting#Depth-first_search) algorithm.

kevinrutherford commented:
Sorry, I don't have time to maintain that code these days (I haven't used
it myself for several years).
Feel free to use the code as the basis for your own gem, however.

Best regards,
Kevin

On Mon, Jan 15, 2018 at 11:54 PM, fsobanski <notifications@github.com>
wrote:

> Would you consider accepting this PR? It's basically just an
> implementation of this
> <https://en.wikipedia.org/wiki/Topological_sorting#Depth-first_search>
> algorithm.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kevinrutherford/dag/pull/12#issuecomment-357815041>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABa-iRqwvYXhhFmvu5Kq3x_MSu-ZuNXks5tK-U5gaJpZM4RLu0Q>
> .
>



-- 

*Kevin Rutherford*
+44 7973 563521
<+44%207973%20563521>
http://rutherford-software.com <http://xpsurgery.com>

Commit:
Try to fix travis rbx build.

See http://www.benjaminfleischer.com/2013/12/05/testing-rubinius-on-travis-ci/

Commit:
Update travis ruby environments

Support of ruby 1.9 and 2.1 has ended:
https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/

Support of Rubinius 2 (rbx-2) has also ended:
https://github.com/rubinius/rubinius/issues/3717#issuecomment-267106474

Commit:
Beautify ruby code with Rubocop

Used Rubocop with default settings as provided by the atom ide.

Commit:
Enforce maximal line length of 80 chars

Commit:
Remove prefix "is" from private predicate methods

See https://github.com/bbatsov/ruby-style-guide#naming
"Avoid prefixing predicate methods with the auxiliary verbs such as is, does, or can. These words are redundant and inconsistent with the style of boolean methods in the Ruby core library, such as empty? and include?."

Commit:
Fix Rubocop linter warning UselessAssignment

Commit:
Remove aliases has_descendent? and has_descendent?

I don't support backwards compatibility.

Commit:
Remove alias is_reachable_from?

I don't support backwards compatibility.

Commit:
Rename has_path_to? to path_to? and has_ancestor? to reachable_from?

See https://github.com/bbatsov/ruby-style-guide#naming
"Avoid prefixing predicate methods with the auxiliary verbs such as is,
does, or can. These words are redundant and inconsistent with the style
of boolean methods in the Ruby core library, such as empty? and include?."

Commit:
Update README

Commit:
Update Copyright notice

Commit:
Simplify method reachable_from?

Commit:
Add CONTRIBUTING.md

Commit:
Add Ruby 2.5 support

Commit:
Store edges in source vertex for performance

Commit:
Let path_to? & reachable_from? return false for vertices in other graph

Commit:
Increase test coverage

Subgraph test should also cover ancestors and descendants more than one
step away.

Commit:
Update Gemfile

Commit:
Revert "Update Gemfile"

The commit broke the travis build for some reason.

This reverts commit 5cc4d5f939e3b720afb0ef1c6788febfd4d45750.

Commit:
Prohibit parallel edges

This is not a multigraph.

Commit:
Add differences to original dag implementation to README
